dsp-value-objects/src/test/scala/dsp/valueobjects/IriSpec.scala

```diff
@@ -0,0 +1,197 @@
+            V2.BadRequestException(IriErrorMessages.ListIriInvalid)
+          )
+        )
+        assertTrue(
+          ListIri.make(Some(invalidIri)) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.ListIriInvalid)
+          )
+        )
+      } +
+      test("pass an invalid IRI containing unsupported UUID version and throw an error") {
+        assertTrue(
+          ListIri.make(listIriWithUUIDVersion3) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.UuidInvalid)
+          )
+        )
+        assertTrue(
+          ListIri.make(Some(listIriWithUUIDVersion3)) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.UuidInvalid)
+          )
+        )
+      } +
+      test("pass a valid value and successfully create value object") {
+        assertTrue(ListIri.make(validListIri).toOption.get.value == validListIri)
+        assertTrue(ListIri.make(Option(validListIri)).getOrElse(null).get.value == validListIri)
+      } +
+      test("pass None") {
+        assertTrue(
+          ListIri.make(None) == Validation.succeed(None)
+        )
+      }
+  )
+
+  private val projectIriTest = suite("IriSpec - ProjectIri")(
+    test("pass an empty value and throw an error") {
+      assertTrue(ProjectIri.make("") == Validation.fail(V2.BadRequestException(IriErrorMessages.ProjectIriMissing)))
+      assertTrue(
+        ProjectIri.make(Some("")) == Validation.fail(V2.BadRequestException(IriErrorMessages.ProjectIriMissing))
+      )
+    } +
+      test("pass an invalid value and throw an error") {
+        assertTrue(
+          ProjectIri.make(invalidIri) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.ProjectIriInvalid)
+          )
+        )
+        assertTrue(
+          ProjectIri.make(Some(invalidIri)) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.ProjectIriInvalid)
+          )
+        )
+      } +
+      test("pass an invalid IRI containing unsupported UUID version and throw an error") {
+        assertTrue(
+          ProjectIri.make(projectIriWithUUIDVersion3) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.UuidInvalid)
+          )
+        )
+        assertTrue(
+          ProjectIri.make(Some(projectIriWithUUIDVersion3)) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.UuidInvalid)
+          )
+        )
+      } +
+      test("pass a valid value and successfully create value object") {
+        assertTrue(ProjectIri.make(validProjectIri).toOption.get.value == validProjectIri)
+        assertTrue(ProjectIri.make(Option(validProjectIri)).getOrElse(null).get.value == validProjectIri)
+      } +
+      test("pass None") {
+        assertTrue(
+          ProjectIri.make(None) == Validation.succeed(None)
+        )
+      }
+  )
+
+  private val UserIriTest = suite("IriSpec - ProjectIri")(
+    test("pass an empty value and throw an error") {
```

Comment:
Overall I'm not sure how the description strings in ZioTest should be phrased

Reply:
Thanks for the info :D